### DISCUSSION TITLE
Caddy to multi instance with different users

### DISCUSSION THREAD

**Discussion by Extarys** [nextcloud/all-in-one#5748] [Questions] [answered]
I've read and re-read the documentation available here, but I feel like something is missing.

I'm using rootless podman with 3 users (userA, userB and userC) for Nextcloud AIO. I've also installed Caddy natively (although I can run it in container if needed).

Since the apache port isn't exposed from the container, caddy cannot reverse-proxy/find the port. For this to work, I would need a caddy container attached to the same network so it could find the port, but then I wouldn't be able to bound 443 to each caddy instance.

The caddy demo shows something like this (https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md):
```
https://usera.domain.local:443 {
    reverse_proxy localhost:11000
}
https://userb.domain.local:443 {
    reverse_proxy localhost:11001
}
https://userc.domain.local:443 {
    reverse_proxy localhost:11002
}
```

- I couldn't find a way to share a network between caddy and the three users
- Some people suggested running all containers under the same user, but the name of the container cannot be renamed so there is a conflict there.

What would be the best approach here ? Is there a downside of exposing port 80 (11000-11002) to the host for Caddy? I would need to disable the `domainchecker` container as it already expose that port.

I'm also wondering, except ssh tunnel, what would be a good way to access the aio webpage (port 8080), without exposing it publicly? My current plan is either ssh tunnel every time, or set up a wireguard vpn.

I look forward to reading the knowledge you'll share.

**Comment by szaimen** [comment]
Hi, can you check https://github.com/nextcloud/all-in-one/blob/main/multiple-instances.md?

**Reply by Extarys** [reply]
Yes, but OnlyOffice couldn't talk to the host (loopback), etc. I redid all my ansible tasks to use docker and it sadly work out of the box. Now I have podman and docker running for different services, yay ! :P

Thanks for chiming in, I appreciate it ;)

**Reply by Extarys** [reply, marked as answer]
Oops, I think I commented for the wrong issue.
Apache listening on ip 0.0.0.0 works fine, I thought I needed to expose the port somehow so caddy could talk to the http port. I didn't do manual docker/podman stuff in a while, I lost the touch. This is working as it is :)
